DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/Repositories/AcademyRepository.cs

```diff
@@ -0,0 +1,22 @@
+using DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Contexts;
+using DfE.FindInformationAcademiesTrusts.Data.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace DfE.FindInformationAcademiesTrusts.Data.AcademiesDb.Repositories;
+
+public class AcademyRepository(IAcademiesDbContext academiesDbContext) : IAcademyRepository
+{
+    public async Task<int> GetNumberOfAcademiesInTrustAsync(string uid)
+    {
+        return await academiesDbContext.GiasGroupLinks.CountAsync(gl => gl.GroupUid == uid && gl.Urn != null);
+    }
+
+    public async Task<string?> GetUrnForSingleAcademyTrustAsync(string uid)
```

Comment:
I might need some context explained to me for this, why only SAT does the trust object just have the URN when it's a SAT as it's a one to one?

Reply:
Renamed to `SingleAcademyTrustAcademyUrn`